### PR TITLE
T25161

### PR DESCRIPTION
**Pull request by dnohales**
https://phabricator.endlessm.com/T25161

### CODE REVIEW

File: convert-sounds.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+sounds_path=$(dirname "$0")/data/sounds
+
+find "$sounds_path" -type f -iregex ".+\\.wav" | while read -r input; do
+    output=${input%.wav}.webm
+    ffmpeg -y -i "${input}" -c:a libopus -compression_level:a 10 -b:a 64k -vn -dash 1 "$output" < /dev/null
```
Comment (manuq) [inline convert-sounds.sh]: `ffmpeg` is not available in the Hack OS. Where is this script supposed to run?

Comment (manuq) [reply convert-sounds.sh]: I think the script should live inside a `./tools/` directory in the root of the repository, for consistency with other hack git repositories.

Comment (cfoch) [reply convert-sounds.sh]: > I think the script should live inside a ./tools/ directory in the root of the repository, for consistency with other hack git repositories.

I agree

Comment (cfoch) [reply convert-sounds.sh]: > `ffmpeg` is not available in the Hack OS. Where is this script supposed to run?

There are some ffmpeg binaries IIRC. I tried it once in Endless.

Comment (manuq) [reply convert-sounds.sh]: Yes `ffmpeg` is inside the Flatpak sandbox, not outside. So this tool should be exposed with Flatpak, and documented in the README.

Comment (ptomato) [reply convert-sounds.sh]: Is there a way to run `ffmpeg` directly with the `flatpak` command line tool?

Comment (cfoch) [reply convert-sounds.sh]: Use `flatpak run --command=ffmpeg org.gnome.Sdk ...`

**Comment by cfoch** [comment]
> I think this script would be great as a git commit hook.
Why as a commit hook for one script?

File: convert-sounds.sh (hunk 1 above)
Comment (cfoch) [reply convert-sounds.sh]: I agree, that is why I suggest that this should be used
```
flatpak run --command=ffmpeg org.gnome.Sdk//3.30 -y -i "${input}" -c:a libopus -compression_level:a 10 -b:a 64k -vn -dash 1 "$output" < /dev/null
```

Comment (dnohales) [reply convert-sounds.sh]: Ok, I'll use the @cfoch approach.

**Comment by dnohales** [comment]
Ok, there's a little problem on using the `ffmpeg` provided by the runtimes available in Hack (well, in any Linux distribution for that matter), basically it was built with no encoders, only decoders:

```
$ flatpak run --command=ffmpeg org.freedesktop.Platform//18.08 -formats
ffmpeg version 4.1 Copyright (c) 2000-2018 the FFmpeg developers
  built with gcc 8.2.0 (GCC)
  configuration: --prefix=/usr --libdir=/usr/lib/x86_64-linux-gnu --disable-debug --disable-doc --disable-static --enable-optimizations --enable-shared --disable-ffplay --disable-ffprobe --disable-everything --enable-gnutls --enable-libfdk-aac --enable-libmp3lame --enable-libfontconfig --enable-libfreetype --enable-libopus --enable-libpulse --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-openal --enable-opengl --enable-sdl2 --enable-encoder='ac3,mp2,libmp3lame,libfdk_aac' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_f32le' --enable-decoder='theora,vorbis,vp8,vp9,mp2,mp3,flac,webp,ac3,libfdk_aac' --enable-decoder='rawvideo,png,gif' --enable-parser='opus,vp3,vorbis,vp8,mpegaudio,flac,aac,ac3' --enable-demuxer='ogg,matroska,wav,mp3,gif,flac,aac,ac3,m4a' --enable-filter='crop,scale' --enable-protocol=file --arch=x86_64
  WARNING: library configuration mismatch
  avutil      configuration: --prefix=/usr/lib/x86_64-linux-gnu/ffmpeg --libdir=/usr/lib/x86_64-linux-gnu/ffmpeg/lib --disable-debug --disable-doc --disable-static --enable-optimizations --enable-shared --disable-ffplay --disable-ffprobe --disable-everything --enable-gnutls --enable-libfdk-aac --enable-libmp3lame --enable-libfontconfig --enable-libfreetype --enable-libopus --enable-libpulse --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-openal --enable-opengl --enable-sdl2 --enable-encoder='ac3,mp2,libmp3lame,libfdk_aac' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_f32le' --enable-decoder='theora,vorbis,vp8,vp9,mp2,mp3,flac,webp,ac3,libfdk_aac' --enable-decoder='rawvideo,png,gif' --enable-parser='opus,vp3,vorbis,vp8,mpegaudio,flac,aac,ac3' --enable-demuxer='ogg,matroska,wav,mp3,gif,flac,aac,ac3,m4a' --enable-filter='crop,scale' --enable-protocol=file --arch=x86_64 --enable-encoder='h263,h264' --enable-hwaccel='h264_vaapi,h264_vdpau' --enable-parser=h264 --enable-demuxer='h264,mov'
  avcodec     configuration: --prefix=/usr/lib/x86_64-linux-gnu/ffmpeg --libdir=/usr/lib/x86_64-linux-gnu/ffmpeg/lib --disable-debug --disable-doc --disable-static --enable-optimizations --enable-shared --disable-ffplay --disable-ffprobe --disable-everything --enable-gnutls --enable-libfdk-aac --enable-libmp3lame --enable-libfontconfig --enable-libfreetype --enable-libopus --enable-libpulse --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-openal --enable-opengl --enable-sdl2 --enable-encoder='ac3,mp2,libmp3lame,libfdk_aac' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_f32le' --enable-decoder='theora,vorbis,vp8,vp9,mp2,mp3,flac,webp,ac3,libfdk_aac' --enable-decoder='rawvideo,png,gif' --enable-parser='opus,vp3,vorbis,vp8,mpegaudio,flac,aac,ac3' --enable-demuxer='ogg,matroska,wav,mp3,gif,flac,aac,ac3,m4a' --enable-filter='crop,scale' --enable-protocol=file --arch=x86_64 --enable-encoder='h263,h264' --enable-hwaccel='h264_vaapi,h264_vdpau' --enable-parser=h264 --enable-demuxer='h264,mov'
  avformat    configuration: --prefix=/usr/lib/x86_64-linux-gnu/ffmpeg --libdir=/usr/lib/x86_64-linux-gnu/ffmpeg/lib --disable-debug --disable-doc --disable-static --enable-optimizations --enable-shared --disable-ffplay --disable-ffprobe --disable-everything --enable-gnutls --enable-libfdk-aac --enable-libmp3lame --enable-libfontconfig --enable-libfreetype --enable-libopus --enable-libpulse --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-openal --enable-opengl --enable-sdl2 --enable-encoder='ac3,mp2,libmp3lame,libfdk_aac' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_f32le' --enable-decoder='theora,vorbis,vp8,vp9,mp2,mp3,flac,webp,ac3,libfdk_aac' --enable-decoder='rawvideo,png,gif' --enable-parser='opus,vp3,vorbis,vp8,mpegaudio,flac,aac,ac3' --enable-demuxer='ogg,matroska,wav,mp3,gif,flac,aac,ac3,m4a' --enable-filter='crop,scale' --enable-protocol=file --arch=x86_64 --enable-encoder='h263,h264' --enable-hwaccel='h264_vaapi,h264_vdpau' --enable-parser=h264 --enable-demuxer='h264,mov'
  avdevice    configuration: --prefix=/usr/lib/x86_64-linux-gnu/ffmpeg --libdir=/usr/lib/x86_64-linux-gnu/ffmpeg/lib --disable-debug --disable-doc --disable-static --enable-optimizations --enable-shared --disable-ffplay --disable-ffprobe --disable-everything --enable-gnutls --enable-libfdk-aac --enable-libmp3lame --enable-libfontconfig --enable-libfreetype --enable-libopus --enable-libpulse --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-openal --enable-opengl --enable-sdl2 --enable-encoder='ac3,mp2,libmp3lame,libfdk_aac' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_f32le' --enable-decoder='theora,vorbis,vp8,vp9,mp2,mp3,flac,webp,ac3,libfdk_aac' --enable-decoder='rawvideo,png,gif' --enable-parser='opus,vp3,vorbis,vp8,mpegaudio,flac,aac,ac3' --enable-demuxer='ogg,matroska,wav,mp3,gif,flac,aac,ac3,m4a' --enable-filter='crop,scale' --enable-protocol=file --arch=x86_64 --enable-encoder='h263,h264' --enable-hwaccel='h264_vaapi,h264_vdpau' --enable-parser=h264 --enable-demuxer='h264,mov'
  avfilter    configuration: --prefix=/usr/lib/x86_64-linux-gnu/ffmpeg --libdir=/usr/lib/x86_64-linux-gnu/ffmpeg/lib --disable-debug --disable-doc --disable-static --enable-optimizations --enable-shared --disable-ffplay --disable-ffprobe --disable-everything --enable-gnutls --enable-libfdk-aac --enable-libmp3lame --enable-libfontconfig --enable-libfreetype --enable-libopus --enable-libpulse --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-openal --enable-opengl --enable-sdl2 --enable-encoder='ac3,mp2,libmp3lame,libfdk_aac' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_f32le' --enable-decoder='theora,vorbis,vp8,vp9,mp2,mp3,flac,webp,ac3,libfdk_aac' --enable-decoder='rawvideo,png,gif' --enable-parser='opus,vp3,vorbis,vp8,mpegaudio,flac,aac,ac3' --enable-demuxer='ogg,matroska,wav,mp3,gif,flac,aac,ac3,m4a' --enable-filter='crop,scale' --enable-protocol=file --arch=x86_64 --enable-encoder='h263,h264' --enable-hwaccel='h264_vaapi,h264_vdpau' --enable-parser=h264 --enable-demuxer='h264,mov'
  swscale     configuration: --prefix=/usr/lib/x86_64-linux-gnu/ffmpeg --libdir=/usr/lib/x86_64-linux-gnu/ffmpeg/lib --disable-debug --disable-doc --disable-static --enable-optimizations --enable-shared --disable-ffplay --disable-ffprobe --disable-everything --enable-gnutls --enable-libfdk-aac --enable-libmp3lame --enable-libfontconfig --enable-libfreetype --enable-libopus --enable-libpulse --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-openal --enable-opengl --enable-sdl2 --enable-encoder='ac3,mp2,libmp3lame,libfdk_aac' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_f32le' --enable-decoder='theora,vorbis,vp8,vp9,mp2,mp3,flac,webp,ac3,libfdk_aac' --enable-decoder='rawvideo,png,gif' --enable-parser='opus,vp3,vorbis,vp8,mpegaudio,flac,aac,ac3' --enable-demuxer='ogg,matroska,wav,mp3,gif,flac,aac,ac3,m4a' --enable-filter='crop,scale' --enable-protocol=file --arch=x86_64 --enable-encoder='h263,h264' --enable-hwaccel='h264_vaapi,h264_vdpau' --enable-parser=h264 --enable-demuxer='h264,mov'
  swresample  configuration: --prefix=/usr/lib/x86_64-linux-gnu/ffmpeg --libdir=/usr/lib/x86_64-linux-gnu/ffmpeg/lib --disable-debug --disable-doc --disable-static --enable-optimizations --enable-shared --disable-ffplay --disable-ffprobe --disable-everything --enable-gnutls --enable-libfdk-aac --enable-libmp3lame --enable-libfontconfig --enable-libfreetype --enable-libopus --enable-libpulse --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-openal --enable-opengl --enable-sdl2 --enable-encoder='ac3,mp2,libmp3lame,libfdk_aac' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_f32le' --enable-decoder='theora,vorbis,vp8,vp9,mp2,mp3,flac,webp,ac3,libfdk_aac' --enable-decoder='rawvideo,png,gif' --enable-parser='opus,vp3,vorbis,vp8,mpegaudio,flac,aac,ac3' --enable-demuxer='ogg,matroska,wav,mp3,gif,flac,aac,ac3,m4a' --enable-filter='crop,scale' --enable-protocol=file --arch=x86_64 --enable-encoder='h263,h264' --enable-hwaccel='h264_vaapi,h264_vdpau' --enable-parser=h264 --enable-demuxer='h264,mov'
  libavutil      56. 22.100 / 56. 22.100
  libavcodec     58. 35.100 / 58. 35.100
  libavformat    58. 20.100 / 58. 20.100
  libavdevice    58.  5.100 / 58.  5.100
  libavfilter     7. 40.101 /  7. 40.101
  libswscale      5.  3.100 /  5.  3.100
  libswresample   3.  3.100 /  3.  3.100
File formats:
 D. = Demuxing supported
 .E = Muxing supported
 --
 D  aac             raw ADTS AAC (Advanced Audio Coding)
 D  ac3             raw AC-3
 D  flac            raw FLAC
 D  gif             CompuServe Graphics Interchange Format (GIF)
 D  h264            raw H.264 video
 D  matroska,webm   Matroska / WebM
 D  mov,mp4,m4a,3gp,3g2,mj2 QuickTime / MOV
 D  mp3             MP2/3 (MPEG audio layer 2/3)
 D  ogg             Ogg
 D  wav             WAV / WAVE (Waveform Audio)
```

**Comment by cfoch** [comment]
What about downloading the binary from https://ffbinaries.com/downloads if ffmpeg is not installed or if it doesn't have the encoders?

the use of this script is just for developers (and we know how to install ffmpeg). After this commit, all the hack-sound-server sounds should be on the webm format. If David sends wav file I (or whoever else) adding the sounds should convert it to webm.

**Comment by dnohales** [comment]
Updated.

**Comment by dnohales** [comment]
BTW, I've reconverted all the sounds since I forgot to use 128k as bitrate.

**Comment by cfoch** [comment]
LGTM, but move it to the tools/ dir, please

**Comment by dnohales** [comment]
@cfoch I did.

**Pull request merged**